From 3760ec6a3922265b23beee00612020742b7dca35 Mon Sep 17 00:00:00 2001
From: Philipp Stephani <phst@google.com>
Date: Wed, 17 Nov 2021 15:13:14 +0100
Subject: [PATCH] Fix entry point discovery on Windows.

The zipfile module always uses forward slashes, so we shouldn't use
os.path.join here.
---
 python/pip_install/extract_wheels/lib/wheel.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git python/pip_install/extract_wheels/lib/wheel.py python/pip_install/extract_wheels/lib/wheel.py
index a60efc6..12993a0 100644
--- python/pip_install/extract_wheels/lib/wheel.py
+++ python/pip_install/extract_wheels/lib/wheel.py
@@ -55,7 +55,10 @@ class Wheel:
             # Calculate the location of the entry_points.txt file
             metadata = self.metadata
             name = "{}-{}".format(metadata.name.replace("-", "_"), metadata.version)
-            entry_points_path = os.path.join("{}.dist-info".format(name), "entry_points.txt")
+            # Note that the zipfile module always uses the forward slash as
+            # directory separator, even on Windows, so don't use os.path.join
+            # here.
+            entry_points_path = "{}.dist-info/entry_points.txt".format(name)
 
             # If this file does not exist in the wheel, there are no entry points
             if entry_points_path not in whl.namelist():
-- 
2.30.1 (Apple Git-130)

